File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.cpp
class Solution {
public:
    bool isVowel(char ch){
        return (ch == 'a') || (ch == 'e') || (ch == 'i') || (ch == 'o') || (ch == 'u');
    }

    string sortVowels(string s) {
        string result;
        vector<char> vowels;
        for(auto ch : s){
            if(isVowel(tolower(ch))) vowels.push_back(ch);
        }
        sort(vowels.begin(), vowels.end());
        int j = 0;
        for(int i = 0; i < s.size(); i++){
            if(!isVowel(tolower(s[i]))) result += s[i];
            else result += vowels[j++];
        }
        return result;
    }
};

File: 2887-sort-vowels-in-a-string/2887-sort-vowels-in-a-string.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = []
        consonant = []
        result = []
        for i in s:
            if i in "aeiouAEIOU":
                vowels.append(i)
            else:
                consonant.append(i)
        vowels.sort()
        i = j = 0
        for alph in s:
            if alph in 'aeiouAEIOU':
                result.append(vowels[i])
                i += 1
            else:
                result.append(consonant[j])
                j += 1
        return "".join(result)
        